stk/calculators/energy/energy_calculators.py

```diff
@@ -377,3 +382,887 @@ def energy(self, mol, conformer=-1):
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        """
+        xtbext = XTBExtrators(output_file=output_file)
+
+        # Get properties from output file.
+        key = (mol, conformer)
+        self.total_energies[key] = xtbext.total_energy()
+        self.homo_lumo_gaps[key] = xtbext.homo_lumo_gap()
+        self.fermi_levels[key] = xtbext.fermi_level()
+        self.homo_lumo_orbitals[key] = xtbext.homo_lumo_occ()
+        self.qonly_dipole_moms[key] = xtbext.qonly_dipole_mom()
+        self.full_dipole_moms[key] = xtbext.full_dipole_mom()
+        self.qonly_quadrupole_moms[key] = xtbext.qonly_quadrupole_mom()
+        self.qdip_quadrupole_moms[key] = xtbext.qdip_quadrupole_mom()
+        self.full_quadrupole_moms[key] = xtbext.full_quadrupole_mom()
+
+    def _write_and_run_command(self, mol, conformer):
+        """
+        Writes and runs the command for xTB.
+
+        Parameters
+        ----------
+        mol : :class:`.Molecule`
+            The :class:`.Molecule` whose energy is to be calculated.
+
+        conformer : :class:`int`, optinal
+            The conformer of `mol` to use.
+
+        Returns
+        -------
+        :class:`str`
+            Name of output file with xTB results.
+
+        """
+        xyz = 'input_structure.xyz'
+        out_file = 'energy.output'
+        mol.write(xyz, conformer=conformer)
+        # Modify the memory limit.
+        if self.unlimited_memory:
+            cmd = ['ulimit -s unlimited ;']
+            # Allow multiple shell commands to be run in one
+            # subprocess.
+            shell = True
+        else:
+            cmd = []
+            shell = False
+        cmd.append(self.xtb_path)
+        cmd.append(xyz)
+        # Set the GFN Parameterization.
+        cmd.append('--gfn')
+        cmd.append(self.gfn_version)
+        # Set the number of cores.
+        cmd.append('--parallel')
+        cmd.append(self.num_cores)
+        # Add eletronic temp term to cmd.
+        cmd.append('--etemp')
+        cmd.append(self.electronic_temperature)
+        # Write the solvent section of cmd.
+        if self.solvent is not None:
+            cmd.append('--gbsa')
+            cmd.append(self.solvent)
+            if self.solvent_grid != 'normal':
+                cmd.append(self.solvent_grid)
+        # Write the charge section of cmd.
+        cmd.append('--chrg')
+        cmd.append(self.charge)
+        # Write the unpaired_electrons section of cmd.
+        cmd.append('--uhf')
+        cmd.append(self.unpaired_electrons)
+
+        cmd = ' '.join(cmd)
+        f = open(out_file, 'w')
+        # Uses the shell if unlimited_memory = True and waits until the
+        # subproces is complete. This is required to be able to run the
+        # unlimited_memory_cmd and GFN calculation in one command,
+        # which is then closed, which minimizes the risk of
+        # unrestricting the memory limits.
```

Comment:
Not following this at all.

Reply:
# Uses the shell if unlimited_memory = True to run multiple commands.